.github/workflows/cross_version_compat.yml

```diff
@@ -0,0 +1,189 @@
+name: Cluster-Client version compatibility 
+
+on:
+  push:
+    branches: [master]
+  workflow_dispatch:
```

Comment:
We need to talk about when we expect to run this. I think tests should be run on `pull_request` or on push to `staging`. Is the expectation that these tests are just data points or would they be prerequisites to merging new code to master?

Reply:
@sehz Do you have an opinion?

I only intended for them to be data points, and I didn't intend for this to communicate stability or performance.

Until we have a concept of a long-term stable version, the value from this test is early warning from `latest`. A sign that we'll need to minor version bump.

I don't think we'll test every version exhaustively, but the `exclude` map does document where we are incompatible, just as a consequence of having the test pass 100%.

Reply:
There are 2 types of workflow we should support:
(1) After every dev release (should not block CI, just warning)
(2) Run locally, in the developers machine

2nd workflow can be treated as separate issue (sort of smoke test).